src/FakeItEasy/Core/FakeManager.PropertySetterRule.cs

```diff
@@ -32,17 +32,31 @@ public void Apply(IInterceptedFakeObjectCall fakeObjectCall)
             {
                 Guard.AgainstNull(fakeObjectCall, nameof(fakeObjectCall));
 
-                var newRule = new CallRuleMetadata
-                                  {
-                                      CalledNumberOfTimes = 1,
-                                      Rule = new PropertyBehaviorRule(fakeObjectCall.Method, this.fakeManager)
-                                                 {
-                                                     Indices = fakeObjectCall.Arguments.Take(fakeObjectCall.Arguments.Count - 1).ToArray(),
-                                                     Value = fakeObjectCall.Arguments.Last()
-                                                 }
-                                  };
-
-                this.fakeManager.AllUserRules.AddFirst(newRule);
+                var existingRule = this.fakeManager
+                    .AllUserRules
+                    .Reverse()
+                    .Select(metadata => metadata.Rule)
+                    .OfType<PropertyBehaviorRule>()
+                    .FirstOrDefault(rule => rule.IsMatchForSetter(fakeObjectCall));

```

Comment:
Oh, yes. `LastOrDefault` and no `Reverse` would be better. And perhaps a comment at the insertion site to indicate why we're shoving the rule at the end instead of the beginning like all the other rules?